### DISCUSSION TITLE
Cache static data

### DISCUSSION THREAD

**Discussion by wgong** [writer/writer-framework#101] [Ideas]
While reading hello app main.py, I wonder how to indicate a function,  data frame, image to be cached, similar to cache_data or cache_resource in streamlit?

Thanks

**Comment by ramedina86** [comment]
Hello, Streamsync doesn't continuously reload the script, so caching isn't needed like in Streamlit. You can use globals or store things in state.

**Comment by wgong** [comment]
I see, all state variables live in memory. thanks for the clarification.

**Comment by FabienArcellier** [comment]
python has also amazing decorator to do exactly what you are doing in ``streamlit`` : [functools.cache](https://docs.python.org/3/library/functools.html#functools.cache) and [functools.lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache).
